lib/quantum-entanglement.js

```diff
@@ -1,49 +1,41 @@
-'use babel';
+/* global atom */
 
-import QuantumEntanglementView from './quantum-entanglement-view';
-import { CompositeDisposable } from 'atom';
+const io = require('socket.io-client')
+const uuidV1 = require('uuid/v1')
 
-export default {
+const { CompositeDisposable } = require('atom')
 
-  quantumEntanglementView: null,
-  modalPanel: null,
-  subscriptions: null,
-
-  activate(state) {
-    this.quantumEntanglementView = new QuantumEntanglementView(state.quantumEntanglementViewState);
-    this.modalPanel = atom.workspace.addModalPanel({
-      item: this.quantumEntanglementView.getElement(),
-      visible: false
-    });
-
-    // Events subscribed to in atom's system can be easily cleaned up with a CompositeDisposable
-    this.subscriptions = new CompositeDisposable();
-
-    // Register command that toggles this view
+module.exports = {
+  activate: function() {
+    this.subscriptions = new CompositeDisposable()
     this.subscriptions.add(atom.commands.add('atom-workspace', {
-      'quantum-entanglement:toggle': () => this.toggle()
-    }));
-  },
-
-  deactivate() {
-    this.modalPanel.destroy();
-    this.subscriptions.dispose();
-    this.quantumEntanglementView.destroy();
+      'atom-plugin-test-aidan:toggle': this.toggle
+    }))
   },
-
-  serialize() {
-    return {
-      quantumEntanglementViewState: this.quantumEntanglementView.serialize()
-    };
+  toggle: function() {
+    const text = atom.workspace.getActiveTextEditor().getBuffer()
+    const socket = io.connect('http://localhost:3000')
+    const listen = () => {
+      if(this.subscription) if(this.subscription.disposalAction) return
+      this.subscription = text.onDidChange(event => {
+        this.lastInstance = uuidV1()
+        socket.emit('edit', Object.assign({}, event, { id: this.lastInstance }))
+      })
+      this.subscriptions.add(this.subscription)
+    }
+    const unlisten = () => {
+      this.subscriptions.remove(this.subscription)
+      this.subscription.dispose()
+    }
+    socket.on('insert', event => {
+      if(this.lastInstance === event.id) return
+      unlisten()
+      text.setTextInRange(event.oldRange, event.newText)
+      listen()
+    })
+    listen()
   },
-
-  toggle() {
-    console.log('QuantumEntanglement was toggled!');
-    return (
-      this.modalPanel.isVisible() ?
-      this.modalPanel.hide() :
-      this.modalPanel.show()
-    );
+  deactivate: function() {
+    this.subscriptions.dispose()
```

Comment:
Before we merge this, can you confirm that you are not spinning up multiple socket connections between toggles?